### PR TITLE
CPS Asset Bold intro

### LINKED ISSUE #4099: MAP introductions should be bold
**Is your feature request related to a problem? Please describe.**
The first paragraph in a MAP should have `role="introduction"` which can be used to add bold rich text mark up to this block. 

**Describe the solution you'd like**
Add a preprocessor rule for MAPs that converts the first paragraph into `bold` rich text. 

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Snapshot testing and E2Es are required to ensure a headline is rendered on the page.

- [x] This feature is expected to need manual testing.

**Additional context**
Example: "She sabi some Nigerian languages and nowadays her fitness regime no be hia" in the following screenshot

image

### PR DESCRIPTION
**Pull request by dr3**
Resolves https://github.com/bbc/simorgh/issues/4099

**Overall change:** Make introductions bold

**Code changes:**

- Adds preprocessor logic to handle bold intros and moves the quote escaping

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export const processBlock = _block => {
+  const block = _block;
+
+  if (path(['text'], block)) {
+    // escape quotes in all text
+    block.text = escapeDoubleQuotes(block.text);
+
+    // make introductions bold
+    if (block.type === 'paragraph' && block.role === 'introduction') {
```
Comment (pjlee11) [inline src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.js]:
```suggestion
    if (block.role === 'introduction') {
```

The `block.type` will always be paragraph by the time we are in the paragraph helper

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.test.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    const expected = {
+      text: 'Paragraph containing "quote marks"',
+      markupType: 'plain_text',
+      type: 'paragraph',
+    };
+
+    expect(processBlock(input)).toEqual(expected);
+  });
+
+  it('should escape quotes', () => {
```
Comment (pjlee11) [inline src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.test.js]:
```suggestion
  it('should escape quotes for candy_xml', () => {
```

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.test.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      text: '<bold><italic>paragraph one</italic></bold>',
+      role: 'introduction',
+      markupType: 'candy_xml',
+      type: 'paragraph',
+    };
+
+    expect(processBlock(input)).toEqual(expected);
+  });
+
+  it('should escape quotes', () => {
```
Comment (pjlee11) [inline src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/paragraph/helpers.test.js]:
```suggestion
  it('should escape quotes for plain_text', () => {
```

**Review by pjlee11 — APPROVED** [review]
LGTM 😎

**Review by jroebu14 — APPROVED** [review]

**Pull request merged**